src/Storages/MergeTree/MergeList.cpp

```diff
@@ -78,6 +67,17 @@ MergeListElement::MergeListElement(
         is_mutation = (result_part_info.getDataVersion() != source_data_version);
     }
 
+    thread_group = std::make_shared<ThreadGroupStatus>();
+
+    auto p_counters = CurrentThread::get().current_performance_counters;
+    while (p_counters && p_counters->level != VariableContext::Process)
+        p_counters = p_counters->getParent();
+    thread_group->performance_counters.setParent(p_counters);
+
+    thread_group->master_thread_id = CurrentThread::get().thread_id;
```

Comment:
This is the only part that looks a bit suspicious.

Actually, I think it's correct. But I don't understand why we did not change performance_counters before (or where we did).

I see the code here
https://github.com/ClickHouse/ClickHouse/blob/6a086ab9264f6a2ce1adefa6f204f76220a9d015/src/Storages/MergeTree/MergePlainMergeTreeTask.cpp#L32
and here
https://github.com/ClickHouse/ClickHouse/blob/6a086ab9264f6a2ce1adefa6f204f76220a9d015/src/Storages/MergeTree/MergePlainMergeTreeTask.cpp#L92

maybe we already don't need it, and can just take counters from the thread group?

Reply:
cc @vdimir

Reply:
Seems like ProfileEventsScope set the counter only for the current thread. 
But we need to configure thread group because the counters from thread group would be used in all task scheduled to thread pools.
Will look more close to this.

Reply:
You can use particular profile_counters in several `ProfileEventsScope`s. The `ProfileEventsScope` should be created and destroyed in the same thread to add metrics from the currently executed function to `profile_counters`, but the same `profile_counters` can be used several times by several threads. But it measures only ProfileCounters with no memory info from MemoryTracker.

Reply:
As I understand, we can use metrics from the newly created thread_group. The only issue is that it may not be created for other tasks. So maybe can do it later.